app/controllers/posts_controller.rb

```diff
@@ -26,10 +26,10 @@ def new
 
   def update
     @post = Post.find(params[:id])
-    if @post.update_attributes(post_params)
+    if @post.update(post_params)
```

Comment:
This is really tricky — see [this post](https://stackoverflow.com/questions/27684038/rails-paperclip-update-vs-update-attributes). In short, `#update` always returns true no matter what, which breaks this conditional. I’m not sure exactly what the solution is, but this code definitely seems not right.